experimental/fluent/agentic/src/test/java/io/serverlessworkflow/fluent/agentic/LC4JEquivalenceIT.java

```diff
@@ -0,0 +1,288 @@
+  }
+
+  @Test
+  @DisplayName("Parallel agents via DSL.parallel(...)")
+  public void parallelWorkflow() {
+    var a1 = AgentsUtils.newFoodExpert();
+    var a2 = AgentsUtils.newMovieExpert();
+
+    Workflow wf = workflow("forkFlow").parallel("fanout", a1, a2).build();
+
+    List<TaskItem> items = wf.getDo();
+    assertThat(items).hasSize(1);
+
+    var fork = items.get(0).getTask().getForkTask();
+    // two branches created
+    assertThat(fork.getFork().getBranches()).hasSize(2);
+    // branch names follow "branch-{index}-{name}"
+    assertThat(fork.getFork().getBranches().get(0).getName()).isEqualTo("branch-0-fanout");
+    assertThat(fork.getFork().getBranches().get(1).getName()).isEqualTo("branch-1-fanout");
+
+    Map<String, Object> input = Map.of("mood", "I am hungry and bored");
+
+    Map<String, Object> result;
+    try (WorkflowApplication app = WorkflowApplication.builder().build()) {
+      result = app.workflowDefinition(wf).instance(input).start().get().asMap().orElseThrow();
+    } catch (Exception e) {
+      throw new RuntimeException("Workflow execution failed", e);
+    }
+
+    assertEquals("Fake conflict response", result.get("meals"));
+    assertEquals("Fake conflict response", result.get("movies"));
+  }
+
+  @Test
+  @DisplayName("Error handling with agents")
+  public void errorHandling() {
+    var a1 = AgentsUtils.newCreativeWriter();
+    var a2 = AgentsUtils.newAudienceEditor();
+    var a3 = AgentsUtils.newStyleEditor();
+
+    Workflow wf = workflow("seqFlow").tasks(tasks -> tasks.sequence("process", a1, a2, a3)).build();
+
+    List<TaskItem> items = wf.getDo();
+    assertThat(items).hasSize(3);
+
+    assertThat(items.get(0).getName()).isEqualTo("process-0");
+    assertThat(items.get(1).getName()).isEqualTo("process-1");
+    assertThat(items.get(2).getName()).isEqualTo("process-2");
+    items.forEach(it -> assertThat(it.getTask().getCallTask()).isInstanceOf(CallTaskJava.class));
+
+    Map<String, Object> input =
+        Map.of(
+            "style", "fantasy",
+            "audience", "young adults");
+
+    Map<String, Object> result;
+    try (WorkflowApplication app = WorkflowApplication.builder().build()) {
+      result = app.workflowDefinition(wf).instance(input).start().get().asMap().orElseThrow();
+    } catch (Exception e) {
+      throw new RuntimeException("Workflow execution failed", e);
+    }
+
+    assertThat(result).containsKey("story");
+  }
+
+  @SuppressWarnings("unchecked")
+  @Test
+  @DisplayName("Conditional agents via choice(...)")
+  public void conditionalWorkflow() {
+
+    var category = AgentsUtils.newCategoryRouter();
+    var a1 = AgentsUtils.newMedicalExpert();
+    var a2 = AgentsUtils.newTechnicalExpert();
+    var a3 = AgentsUtils.newLegalExpert();
+
+    Workflow wf =
+        workflow("conditional")
+            .sequence("process", category)
+            .tasks(
+                t ->
+                    t.switchCase(
+                        p ->
```

Comment:
We don't need this branch. On `UNKNOWN` the workflow just won't call any agent. What you can do is a `when(UNKNOWN).emit(eventAlertUnknown)`.